apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -0,0 +1,565 @@
+import { escapeHtml, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { Modal } from '../../components/Modal.html';
+import { IdSchema } from '../../lib/db-types';
+import { CourseWithPermissions } from '../../models/course';
+import { isEnterprise } from '../../lib/license';
+
+export const SelectedAssessmentsSchema = z.object({
+  title: z.string(),
+  course_instance_id: IdSchema,
+  assessments: z.array(
+    z.object({
+      assessment_id: IdSchema,
+      color: z.string(),
+      label: z.string(),
+    }),
+  ),
+});
+type SelectedAssessments = z.infer<typeof SelectedAssessmentsSchema>;
+
+export const SharingSetSchema = z.object({
+  id: IdSchema,
+  name: z.string(),
+  in_set: z.boolean(),
+});
+type SharingSet = z.infer<typeof SharingSetSchema>;
+
+export function InstructorQuestionSettings({
+  resLocals,
+  questionTestPath,
+  questionTestCsrfToken,
+  questionGHLink,
+  qids,
+  assessmentsWithQuestion,
+  sharingEnabled,
+  sharingSetsIn,
+  sharingSetsOther,
+  editableCourses,
+  infoPath,
+}: {
+  resLocals: Record<string, any>;
+  questionTestPath: string;
+  questionTestCsrfToken: string;
+  questionGHLink: string | null;
+  qids: (string | null)[];
```

Comment:
When will this array contain null elements?

Reply:
Removed `null` in [e66ab28](https://github.com/PrairieLearn/PrairieLearn/pull/9571/commits/e66ab286e419ca64e565b7f65a79937d66b73331).

Reply:
Seems like you also need to update a corresponding Zod type.

Reply:
Good catch. Updated Zod type in [859ec79](https://github.com/PrairieLearn/PrairieLearn/pull/9571/commits/859ec79031dbc9c954fdada6e61b53a6dc00f3f7).